packages/fxa-payments-server/src/routes/Product/index.tsx

```diff
@@ -205,131 +172,108 @@ export const Product = ({
+    if (customerIsSubscribedToProduct(customerSubscriptions, productPlans)) {
+      return <SubscriptionRedirect {...{ plan: selectedPlan }} />;
+    }
   }
 
-  const inProgress = createSubscriptionStatus.loading;
-
   return (
-    <div className="product-payment">
-      {profile.result && (
-        <>
-          {accountActivated ? (
-            <AccountActivatedBanner profile={profile.result} />
-          ) : (
-            <ProfileBanner profile={profile.result} />
-          )}
-          <hr />
-        </>
-      )}
-
-      <PlanDetails plan={selectedPlan} />
-
-      <hr />
-
-      <h3 className="billing-title">
-        <span>Billing Information</span>
-      </h3>
-
-      <ErrorMessage isVisible={!!createTokenError.error}>
-        {createTokenError.error && (
-          <p data-testid="error-payment-submission">
-            {getErrorMessage(createTokenError.type)}
-          </p>
-        )}
-      </ErrorMessage>
-
-      <ErrorMessage isVisible={isCardError}>
-        <p data-testid="error-card-rejected">{getErrorMessage('card_error')}</p>
-      </ErrorMessage>
-
-      {createSubscriptionStatus.error && !isCardError && (
-        <DialogMessage
-          className="dialog-error"
-          onDismiss={resetCreateSubscriptionError}
-        >
-          <h4 data-testid="error-subscription-failed">Subscription failed</h4>
-          <p>{createSubscriptionStatus.error.message}</p>
-        </DialogMessage>
-      )}
-
-      <PaymentForm
-        {...{
-          onPayment,
-          onPaymentError,
-          onChangeErrorDismiss,
-          inProgress,
-          validatorInitialState,
-          confirm: true,
-          plan: selectedPlan,
-          onMounted: createSubscriptionMounted,
-          onEngaged: createSubscriptionEngaged,
-        }}
-      />
-    </div>
+    <SubscriptionCreate
+      {...{
+        profile: profile.result,
+        accountActivated,
+        selectedPlan,
+        createSubscriptionAndRefresh,
+        createSubscriptionStatus,
+        resetCreateSubscription,
+        validatorInitialState,
+        createSubscriptionMounted,
+        createSubscriptionEngaged,
+      }}
+    />
   );
 };
 
-type ProfileProps = {
-  profile: Profile;
+type PlansByIdType = {
+  [plan_id: string]: { plan: Plan; metadata: ProductMetadata };
 };
 
-const ProfileBanner = ({
-  profile: { email, avatar, displayName },
-}: ProfileProps) => (
-  <div className="profile-banner">
-    <img className="avatar hoisted" src={avatar} alt={displayName || email} />
-    {displayName && (
-      <h2 data-testid="profile-display-name" className="displayName">
-        {displayName}
-      </h2>
-    )}
-    <h3 data-testid="profile-email" className="name email">
-      {email}
-    </h3>
-    {/* TODO: what does "switch account" do? need to re-login and redirect eventually back here?
-      <a href="">Switch account</a>
-    */}
-  </div>
-);
+const indexPlansById = (plans: PlansFetchState): PlansByIdType =>
```

Comment:
Could probably move this into `store/selectors.ts`, but I was also considering moving the `plansByProductId` selector *out* of there.